catt/controllers.py

```diff
@@ -35,8 +41,37 @@ def get_chromecast(device_name):
         return devices[0]
 
 
-def human_time(seconds):
-    return time.strftime("%H:%M:%S", time.gmtime(seconds))
+def setup_cast(device_name, video_url=None, prep=None):
+    cache = Cache()
+    cached_ip = cache.get(device_name)
+
+    try:
+        if not cached_ip:
+            raise ValueError
+        cast = pychromecast.Chromecast(cached_ip)
+    except (pychromecast.error.ChromecastConnectionError, ValueError):
+        cast = get_chromecast(device_name)
+        cache.set(cast.name, cast.host)
+    cast.wait()
+
+    if video_url:
+        cc_info = (cast.device.manufacturer, cast.model_name)
+        stream = StreamInfo(video_url, model=cc_info, host=cast.host)
+        try:
+            if stream.is_local_file:
+                raise ValueError
+            app_id = APP_IDS[stream.extractor]
+        except (KeyError, ValueError):
+            app_id = DEFAULT_APP_ID
+    else:
+        stream = None
+        app_id = cast.app_id
+
```

Comment:
I know this is a bit creative, but it seemed like the least weird solution. At one point I considered putting the `stream` object inside the controller object, but decided against it, as I think this solution makes the `CastController` classes themselves more concise. I'd also like to preserve the explicit nature of current `cli.py`.